From 53908c0a23a57927b7181746e4cd0f8bff6ba84d Mon Sep 17 00:00:00 2001
From: maxwen <max.weninger@gmail.com>
Date: Sat, 20 Jan 2018 13:39:17 +0100
Subject: [PATCH 07/11] SimSettings: improve dialogs

Squash of:

Settings: no need to show a success alert dialog on sim switch

very 'un-android-like' :)

Change-Id: Ie68eed174464d7acf67171a509a19045e0d9c669

Settings: fix neverending progress on sim activation change

what a stupid bug....

and while we are here - also the confirm dialog on
deactivation is actually useless - using the switch
should be more enough to not always ask and anoy user
if he really knows what he is doing

Change-Id: I38eb17cf8d12259f34b339da2ad807d12a5c0d41
---
 src/com/android/settings/sim/SimSettings.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/com/android/settings/sim/SimSettings.java b/src/com/android/settings/sim/SimSettings.java
index 2d71c0c..275f9a8 100644
--- a/src/com/android/settings/sim/SimSettings.java
+++ b/src/com/android/settings/sim/SimSettings.java
@@ -608,7 +608,7 @@ public class SimSettings extends RestrictedSettingsFragment implements Indexable
             if (!mIsChecked) {
                 if (getNumOfSubsProvisioned() > 1) {
                     logd("More than one sub is active, Deactivation possible.");
-                    showAlertDialog(CONFIRM_ALERT_DLG_ID, 0);
+                    sendUiccProvisioningRequest();
                 } else {
                     logd("Only one sub is active. Deactivation not possible.");
                     showAlertDialog(ERROR_ALERT_DLG_ID, R.string.sim_enabler_both_inactive);
@@ -809,7 +809,7 @@ public class SimSettings extends RestrictedSettingsFragment implements Indexable
                                 showAlertDialog(ERROR_ALERT_DLG_ID, msgId);
                             } else {
                                 mCurrentUiccProvisionState = newProvisionedState == PROVISIONED;
-                                showAlertDialog(RESULT_ALERT_DLG_ID, 0);
+                                cleanUpPendingDialogs();
                             }
                             mHandler.removeMessages(EVT_PROGRESS_DLG_TIME_OUT);
                             break;
-- 
2.7.4

